lib/storage.js

```diff
@@ -16,31 +16,44 @@ const defaultResponse = {
   removeOverlay: false
 };
 
+const DEFAULT_LOCALE = 'en';
+
 export function getSettings() {
-  return new Promise((resolve) => {
-    getStorage()
-      .read()
-      .then((data) => {
-        if (!data.settings) {
-          return resolve(defaultResponse);
-        }
-
-        return resolve(data.settings);
-      })
-      .catch(() => resolve(defaultResponse));
-  });
+  return getStorage()
+    .read()
+    .then((data) => {
```

Comment:
This function could be shortened to `data => data.settings || defaultSettingsResponse`